### ISSUE TITLE
Fix corrupts the use of "emoji" (and the like) symbols

### ISSUE THREAD

**Issue by EdmundsEcho** [sqlfluff/sqlfluff#1332]
The fix function changes the, strictly speaking, non-ascii spec text.

## Expected Behaviour
Do nothing to the text in the comment section; at minimum, do not change.  So I would expect:
#### before fix
--  👉 

#### after fix
-- 👉 

#### before fix
-- 👉 

#### after fix
-- â ï¸\x8f

## Steps to Reproduce
```
sqlfluff fix myfile.sql
```

## Dialect
ansi

## Version
Include the output of `sqlfluff --version` along with your Python version
python: 3.9
fluff version: 0.6.4

## Configuration
None

**Comment by tunetheweb** [comment]
Can't reproduce. You sure you had the encoding correct on the file?

Can you give a fully test case with something that requires fixing? Cause just a comment won't cause the fix to do anything.

**Comment by EdmundsEcho** [comment]
It's a good question.  The encoding is utf-8 (returned using both fileencoding and encoding commands in nvim). 

What I can say is that I've run automated fixes in my code in other languages all using neovim, and these glyphs without experiencing what I have described.  I use mac os. The sqlfluff fix was executed in the terminal.

Is there something else I can report to identify the state that generates the issue?

**Comment by EdmundsEcho** [comment]
This is not a direct hit but hopefully you can make this more so.

Here is the code that I tested:
```sql
--------------------------------------------------------------------------------
-- 🔑 schema usage and object (table) permissions
-- required for the sessions table
-- auth: needs a view of users to validate their credentials
grant usage on schema core to auth;
grant references on table core.users to auth;

--------------------------------------------------------------------------------
-- 📖 Sessions
-- indexed using token; informs the active_users view
--
create TABLE sessions (
    token text not null primary key default encode(gen_random_bytes(32), 'base64'),
    user_id uuid not null references core.users,  -- schema bc can't ref fk
    created timestamptz not null default clock_timestamp(),
    expires timestamptz not null default clock_timestamp() + '15min' ::interval,
    check (expires > created)
);
COMMENt on table sessions is 'User sessions, both active and expired ones.';
comment on column sessions.expires is 'Time on which the session expires.';
```

Here is the lint report:
```fish
sqlfluff lint ./schema/tmp.sql
== [schema/tmp.sql] FAIL
L:  12 | P:   1 |  PRS | Line 12, Position 1: Found unparsable section: 'create
                       | TABLE sessions (\n token text n...'
L:  19 | P:   1 |  PRS | Line 19, Position 1: Found unparsable section: "COMMENt
                       | on table sessions is 'User sessi..."
L:  20 | P:   1 |  PRS | Line 20, Position 1: Found unparsable section: "comment
                       | on column sessions.expires is 'T..."
All Finished 📜 🎉!
```

The fix command is not changing this file... do you have recommendation to get a response you know should work:

```fish
sqlfluff fix ./schema/tmp.sql
==== finding fixable violations ====
==== no fixable linting violations found ====
All Finished 📜 🎉!
```

**Comment by tunetheweb** [comment]
If you have a file called test.sql with this in it:

```sql
-- 👉
select a, b
from table_a
```

And run this, with no `.sqlfluff` config file:
```
sqlfluff fix -f --fixed-suffix FIXED test.sql
```

It should create a `testFIXED.sql` file, which for me has the following in it:
```sql
-- 👉
select
    a,
    b
from table_a
```

Hence I can't reproduce your issue 🤷‍♂️

**Comment by EdmundsEcho** [comment]
Funny stuff.  I have since tried to reproduce it and cannot.  Apologies for my mistake.
